godot-core/src/builtin/file.rs

```diff
@@ -0,0 +1,459 @@
+        self.fa.store_line(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_csv_line(
+        &mut self,
+        values: PackedStringArray,
+        delim: GodotString,
+    ) -> IoResult<()> {
+        self.fa.store_csv_line_ex(values).delim(delim).done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_variant(&mut self, value: Variant, full_objects: bool) -> IoResult<()> {
+        self.fa
+            .store_var_ex(value)
+            .full_objects(full_objects)
+            .done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    // Varia
+    pub fn set_big_endian(&mut self, value: bool) {
+        self.fa.set_big_endian(value);
+    }
+
+    pub fn is_big_endian(&self) -> bool {
+        self.fa.is_big_endian()
+    }
+
+    pub fn get_path(&self) -> GodotString {
+        self.fa.get_path()
+    }
+
+    pub fn get_path_absolute(&self) -> GodotString {
+        self.fa.get_path_absolute()
+    }
+
+    /// Returns the current file position
+    pub fn get_position(&self) -> u64 {
+        self.fa.get_position()
+    }
+}
+
+impl Read for GFile {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        let length = self.fa.get_length();
+        let position = self.fa.get_position();
+
+        if position >= length {
+            return Ok(0);
+        }
+
+        let remaining_bytes = (length - position) as usize;
+        let bytes_to_read = cmp::min(buf.len(), remaining_bytes);
+
+        if bytes_to_read == 0 {
+            return Ok(0);
+        }
+        let mut readen = 0;
+        while readen < bytes_to_read {
+            buf[readen] = self.fa.get_8();
+            readen += 1;
+            self.check_error()?;
+        }
+        Ok(readen)
+    }
+}
+
+impl Write for GFile {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let bytes_to_write = buf.len();
+        let mut bytes_written = 0;
+
+        while bytes_written < bytes_to_write {
+            self.fa.store_8(buf[bytes_written]);
+            bytes_written += 1;
+            self.check_error()?;
+        }
+        Ok(bytes_written)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+impl Seek for GFile {
+    fn seek(&mut self, pos: std::io::SeekFrom) -> std::io::Result<u64> {
+        match pos {
+            std::io::SeekFrom::Start(position) => {
+                self.fa.seek(position);
+                Ok(position)
+            }
+            std::io::SeekFrom::End(position) => {
+                if (self.fa.get_length() as i64) < position {
+                    return Err(IoError::new(
+                        ErrorKind::InvalidInput,
+                        "Position can't be set before the file beginning",
+                    ));
+                }
+                self.fa.seek_end_ex().position(position).done();
+                Ok(self.fa.get_position())
+            }
```

Comment:
I would name this `offset` and not `position`, to avoid confusion with the method `position` that gives the absolute position from the start.

Same for `Current(...)` variant.